torch/distributed/distributed_c10d.py

```diff
@@ -1179,12 +1179,19 @@ def send(tensor: torch.Tensor, dst: int, group: Optional[ProcessGroup] = None, t
 
     Args:
         tensor (Tensor): Tensor to send.
-        dst (int): Destination rank.
+        dst (int): Destination rank. Destination rank should not be the same
+        as the rank of the current process.
         group (ProcessGroup, optional): The process group to work on. If None,
             the default process group will be used.
         tag (int, optional): Tag to match send with remote recv
 
     """
+    if get_rank() == dst:
+        raise RuntimeError(
```

Comment:
This might be more descriptive as a `ValueError` instead? https://docs.python.org/3/library/exceptions.html#ValueError

Reply:
Good catch! Will update.